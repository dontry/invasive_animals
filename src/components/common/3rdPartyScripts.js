import React from "react";
import Script from "react-load-script";

export const TableauScript = () => {
  return (
    <Script
      url="https://public.tableau.com/javascripts/api/viz_v1.js"
      onCreate={() => {
        console.log("create script");
      }}
      onError={() => {
        console.log("load script error");
      }}
      onLoad={() => {
        console.log("load tableau script");
      }}
    />
  );
};
