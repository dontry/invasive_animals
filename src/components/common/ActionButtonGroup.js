import React from "react";
import PropTypes from "prop-types";
import Button from "material-ui/Button";
import styled from "styled-components";

const ButtonGroupWrapper = styled.div`
  text-align: center;
`;

export const StyledButton = styled(Button)`
  && {
    margin-left: 5px;
    margin-right: 5px;
    background-color: ${props =>
      props.theme.palette
        ? props.type
          ? props.trait
            ? props.theme.palette[props.type][props.trait]
            : props.theme.palette[props.type].main
          : props.theme.primary.main
        : "#fff"};
    color: ${props => props.theme.palette ? props.theme.palette[props.type].contrastText : "#222"};
    font-size: ${props => props.theme.palette ? props.theme.text_size.size : "1em"};
    height: ${props => props.height || "auto"};
    width: ${props => props.width || "auto"};
    padding ${props => props.padding || "auto"};

    &:hover {
      background-color: ${props => props.theme.palette ? props.theme.palette[props.type].dark : "#acacac"};
    }
  }
`;

export const ActionButton = ({
  raised,
  action,
  label,
  height,
  width,
  padding,
  type = "primary",
  trait = "main"
}) => (
  <StyledButton
    variant={raised ? "raised" : "flat"}
    onClick={action}
    type={type}
    trait={trait}
    height={height}
    width={width}
    padding={padding}
  >
    {label}
  </StyledButton>
);

const ActionButtonGroup = ({ btnStyle, primaryProps, secondaryProps }) => {
  return (
    <ButtonGroupWrapper btnStyle={btnStyle}>
      <ActionButton className="action-btn" {...primaryProps} />
      <ActionButton className="action-btn" {...secondaryProps} />
    </ButtonGroupWrapper>
  );
};

ActionButtonGroup.propTypes = {
  style: PropTypes.object.isRequired,
  primaryProps: PropTypes.shape({
    label: PropTypes.oneOfType([PropTypes.string, PropTypes.element]),
    primary: PropTypes.bool,
    onClick: PropTypes.func,
    disabled: PropTypes.bool,
    type: PropTypes.string
  }),
  secondaryProps: PropTypes.shape({
    label: PropTypes.oneOfType([PropTypes.string, PropTypes.element]),
    primary: PropTypes.bool,
    onClick: PropTypes.func,
    disabled: PropTypes.bool,
    type: PropTypes.string
  })
};

ActionButtonGroup.defaultProps = {
  style: {},
  primaryProps: {
    raised: true,
    label: "Primary",
    primary: false,
    action() {
      console.log("onClick");
    },
    disabled: false,
    type: "primary",
    trait: "main"
  },
  secondaryProps: {
    label: "Secondary",
    raised: false,
    primary: false,
    action() {
      console.log("onClick");
    },
    disabled: false,
    type: "secondary",
    trait: "light"
  }
};

export default ActionButtonGroup;
