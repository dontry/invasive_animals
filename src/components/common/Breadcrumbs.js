import React, { Children } from "react";
import PropTypes from "prop-types";
import { grey } from "material-ui/colors";
import styled from "styled-components";
import { Link } from "react-router-dom";
import NavigationLink from "./NavLink";
import Icon from "material-ui/Icon";
import { findRouteName } from "../../utils/routing";

export const Separator = styled.span`
  display: inline-block;
  color: ${props => props.color || grey[400]};
  text-align: center;
  flex-shrink: 0;
  padding: ${props => props.padding || "0.5rem"};
`;

Separator.displayName = "Separator";

export const BreadcrumbsItem = ({match,  hasSeparator = false }) => {
  let routeName = findRouteName(match.url);
  let content = null;
  if (routeName) {
    routeName = routeName !== "HOME" ? routeName : <Icon style={{ verticalAlign: "-0.2rem" }}>home</Icon> 
    return (
      <span>
        <NavigationLink textSize="small" to={match.url || ''}>{routeName}</NavigationLink>
        {hasSeparator && <Separator>/</Separator>}
      </span>
    );
  }
};

BreadcrumbsItem.propTypes = {
  hasSeparator: PropTypes.bool
};

//Atlakit breadcrumb
const { toArray } = Children;

export const Breadcrumbs = ({ children }) => {
  const allNonEmptyItems = toArray(children);
  return allNonEmptyItems.map((child, index) =>
    React.cloneElement(child, {
      hasSeparator: index < children.length - 1
    })
  );
};
