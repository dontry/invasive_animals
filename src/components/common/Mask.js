import styled from "styled-components";

export const Mask = styled.div`
  position: absolute;
  width: 100%;
  height: 100%;
  &:after {
    position: absolute;
    content: "";
    z-index: 10;
    top: 0;
    bottom: 0;
    left: 0;
    right: 0;
    background-color: rgba(255, 255, 255, ${props => props.opacity || 0.5});
  }
`;

export const ScreenMask = Mask.extend`
  position: fixed;
  width: 100vw;
  height: 100vh;
  z-index: 10;
  background: transparent;
  &:after {
    position: fixed;
    content: "";
  }
`;
