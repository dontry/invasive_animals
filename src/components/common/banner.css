.slick-dots {
    bottom: 25px;
}


.slick-dots li button::before {
    font-size: 12px;
    color: rgba(217, 243, 186, 1);
}

.slick-dots li.slick-active button::before {
    color: rgb(139, 195, 74);
}