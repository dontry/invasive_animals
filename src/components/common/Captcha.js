import React, {Fragment} from "react";
import PropTypes from "prop-types";
import ReCAPTCHA from "react-google-recaptcha";
import { red } from "material-ui/colors";
import { Paragraph } from "../common/Text";


const Captcha = ({ input, meta: { touched, error } }) => (
  <Fragment>
    <Paragraph text_color={red[500]}>{touched && error}</Paragraph>
    <ReCAPTCHA
      sitekey="6LfMElYUAAAAAH8NaQnWfmsegkiaRjFqZ9Qq9ILi"
      onChange={res => input.onChange(res)}
    />
  </Fragment>
);

Captcha.propTypes = {
  input: PropTypes.object,
  meta: PropTypes.object
}

export default Captcha;