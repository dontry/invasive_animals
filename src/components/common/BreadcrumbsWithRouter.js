import React, { Component } from "react";
import { Route, withRouter } from "react-router-dom";
import { Breadcrumbs, BreadcrumbsItem, Separator } from "./Breadcrumbs";

function renderBreadcrumbItems(paths) {
  return paths.map((p, index) => {
    if (index < paths.length - 1) {
      return (
        <span>
          <Route key={p} path={p} component={BreadcrumbsItem} />
          <Separator>/</Separator>
        </span>
      );
    } else {
      return <Route key={p} path={p} component={BreadcrumbsItem} />;
    }
  });
}

const BreadcrumbsWithRouter = ({ location, match }) => {
  const paths = [];
  location.pathname.split("/").reduce((prev, curr, index) => {
    paths[index] = `${prev}/${curr}`;
    return paths[index];
  });
  if(paths[1] !== "/") paths[0] = "/";
  const breadcrumbsItems = renderBreadcrumbItems(paths);
  return <Breadcrumbs>{breadcrumbsItems}</Breadcrumbs>;
};

export default withRouter(BreadcrumbsWithRouter);
