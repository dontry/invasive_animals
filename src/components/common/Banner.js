import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
//Material UI
import Grid from "material-ui/Grid";

import Slider from "react-slick";
import "./banner.css";

const StyledSlider = styled(Slider)`
`;

const BannerWrapper = styled.div`
  position: relative;
  box-sizing: border-box;
  min-height: 300px;
  height: 45vh;
  width: 100%;
  z-index: 1;
  &::before {
    content: "";
    position: absolute;
    margin: ${props => (props.padding ? "-" + props.padding : "-1rem")};
  }
  @media screen and (max-width: 400px) {
    & {
      height: 30vh;
    }
  }
`;

const BackgroundImage = styled.img`
  position: absolute;
  width: 100%;
  height: 100%;
  z-index: -10;
  filter: brightness(80%);
`;

const BlurbWrapper = styled(Grid)`
  && {
    padding-top: 1rem;
    padding-left: 2rem;
  }
`;

const Banner = ({ timeout = 0, heading, description, banners, textColor }) => {
  const settings = {
    dots: true,
    infinite: true,
    lazyLoad: true,
    fade: true,
    autoplay: true,
    speed: 5000,
    slidesToShow: 1,
    slidesToScroll: 1,
    arrows: false
  };
  return (
    <StyledSlider {...settings}>
      {banners.map((banner, index) => (
        <BannerWrapper key={`banner-${index}`}>
          <BackgroundImage src={banner.image} />
          <BlurbWrapper
            container
            direction="column"
            justify="center"
            alignItems="flex-start"
          >
            <Grid item xs={12} md={9}>
              {banner.description}
            </Grid>
          </BlurbWrapper>
        </BannerWrapper>
      ))}
    </StyledSlider>
  );
};

Banner.propTypes = {
  heading: PropTypes.string.isRequired,
  subheading: PropTypes.string,
  description: PropTypes.string,
  imgUrl: PropTypes.string
};

Banner.defaultProps = {
  heading: "",
  description: "",
  imgUrl: "",
  textColor: "#fff"
};

export default Banner;
