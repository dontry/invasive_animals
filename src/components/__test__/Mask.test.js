import React from "react";
import { configure, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { Mask } from "../../components/common/Mask";
import toJson from "enzyme-to-json";

configure({ adapter: new Adapter() });
describe("<Mask />", () => {
  it("should render Mask", () => {
    const wrapper = shallow(<Mask />, {
      context: {},
      disableLifeCycleMethods: false
    });
    console.log(wrapper.debug());
    expect(wrapper.find("div").length).toBe(1);
  });

  //Generate snapshot
  it("matches the snapshot", () => {
    const tree = shallow(<Mask />);
    expect(toJson(tree)).toMatchSnapshot();
  });
});
