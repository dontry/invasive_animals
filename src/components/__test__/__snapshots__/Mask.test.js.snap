// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Mask /> matches the snapshot 1`] = `
.c0 {
  position: absolute;
  width: 100%;
  height: 100%;
}

.c0:after {
  position: absolute;
  content: "";
  z-index: 10;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  background-color: rgba(255,255,255,0.5);
}

<div
  className="c0"
/>
`;
